go/adbc/driver/snowflake/record_reader.go

```diff
@@ -219,7 +219,7 @@ func getTransformer(sc *arrow.Schema, ld gosnowflake.ArrowStreamLoader) (*arrow.
 		default:
 			transformers[i] = identCol
 		}
-
+		f.Name = strings.ToLower(f.Name)
```

Comment:
Hmm, do we want to unconditionally do this?

Reply:
I'm not sure. Unfortunately we're probably gonna run into this more as we add more drivers. Snowflake currently returns *everything* unconditionally upper-cased as far as I can tell. While our tests expect things to be lower cased. I personally prefer consistency so I like the idea of saying we lower case the field names..... but I dunno.

Reply:
Hmm, I'd rather return what the database gives us unchanged as much as possible. The validation suite will need to get more flexible.

Reply:
Okay, i'll leave it returning upper case and then adjust CompareSchema to check case-insensitively